[D2] ListCardView, DeadLineView UI 구현

kimkyunghun3 opened the pull request:
# 개요
- 🔗  이슈링크 : #5 

## 작업 내용
ListCardView UI 구현 ( Font, Color는 리베이스를 아직 하지 못해서 적용이 되지 않았는데 리베이스 해야할지 의논 후 적용하겠습니다 )
DeadLineView UI 구현

### 스크린샷

<img src ="https://user-images.githubusercontent.com/52434820/180615607-473e389e-3e45-4426-b0ef-fc1086d9024b.png" width ="400">|<img src ="https://user-images.githubusercontent.com/52434820/180632118-da295fe3-83fc-4e4f-b227-7e7ad36e62cf.png" width ="400">|

kimkyunghun3 committed:
:bulb: DeadlineViewStyle 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: deadlineView 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: StackView 추가하는 Sugar extension 추가

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: DeadlineViewStyle 주입받는 부분 제거

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: viewer, CTAButton 주입받는 부분 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: ListCardViewStyle 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: ListCardView 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: viewer 불필요한 코드 삭제 및 필요 할당 값 적용

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListViwer 내부 짤리지 않도록 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: CardListButton isHidden 프로퍼티 추가

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListViwer inset 추가 및 레이아웃 수정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListCardView stackView hidden에 따라 이동하도록 변경 

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 내부 요소들 생성자 방식 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: ListCardView 절취선 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: 절취선 양옆에 Space 표시 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: space 반원 그리도록 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 점선 그리는 extension 파일 분리

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: spaceView height 조정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: spaceView 구현 및 생성자 방식 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: Viwer 레이아웃 수정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: TimeView 로직 변경 및 레이아웃 수정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 레이아웃별 분리

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: Date 변환 로직 sugar code 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: TimeView cornerRadius 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

AhnSangHee commented on Projects/DesignSystem/Sources/Component/Button/DDIPCardListButton.swift:
```diff
@@ -27,6 +27,7 @@ public class DDIPCardListButton: UIButton {
         self.setTitleColor(self.style.titleColor, for: .normal)
         self.backgroundColor = self.style.buttonColor
+        self.isHidden = self.style.isHidden
         self.titleLabel?.font = .systemFont(ofSize: 14)
```
🔴 디자인 시스템 폰트를 사용해주시면 감사하겠습니다!🙂

kimkyunghun3 replied:
이 부분은 다른 컴포넌트도 적용해야해서 PR 이후 머지되면 전체적으로 적용해서 다시 PR 올리겠습니다!

AhnSangHee commented on Projects/DesignSystem/Sources/Component/DeadlineCard/DDIPDeadlineView.swift:
```diff
@@ -0,0 +1,228 @@
+    }
+    
+    private func setFont() {
+        nameLabel.font = UIFont.systemFont(ofSize: 18)
+        brandLabel.font = UIFont.systemFont(ofSize: 12)
+        expirationLabel.font = UIFont.systemFont(ofSize: 12)
```
🔴 요기도 디자인 시스템 폰트를 사용해주세요! 😀

kimkyunghun3 replied:
이 부분은 다른 컴포넌트도 적용해야해서 PR 이후 머지되면 전체적으로 적용해서 다시 PR 올리겠습니다!

AhnSangHee commented on Projects/DesignSystem/Sources/Component/Extension/StackView+.swift:
```diff
@@ -0,0 +1,17 @@
+//
+//  StackView+.swift
```
🟡 별건 아니지만 `StackView+` 대신 `UIStackView+`로 네이밍해도 좋을 것 같습니다!

kimkyunghun3 replied:
넵 더 명확하게 바꾸었습니다!!!

AhnSangHee commented on Projects/DesignSystem/Sources/Component/Extension/UIView+sugar.swift:
```diff
@@ -0,0 +1,23 @@
+        let caShapeLayer = CAShapeLayer()
+        caShapeLayer.strokeColor = color
+        caShapeLayer.lineWidth = width
+        caShapeLayer.lineDashPattern = [2,4]
```
❓ [2, 4]에서 2가 페인팅된 세그먼트, 4가 페인팅되지 않은 세그먼트라고 생각이 되는데 혹시 맞을까요?!

만약 맞다면! 저희 피그마를 보면 점선이 페인팅된 세그먼트와 페인팅되지 않은 세그먼트가 동일한 너비를 갖는 걸로 알고 있는데 [2, 4]가 아닌 [2, 2] 또는 [4, 4]로 값을 줘야하지 않나 하는 생각이 듭니다!!

kimkyunghun3 replied:
앗 확인해보니 3.5씩 되어있어서 3.5 3.5로 수정했습니다 감사합니다!

AhnSangHee commented on Projects/DesignSystem/Sources/Component/DeadlineCard/DDIPDeadlineView.swift:
```diff
@@ -0,0 +1,228 @@
+public class DDIPDeadlineView: UIView, AddViewsable {
+    public let style: DDIPDeadlineViewStyle
+    private let CTAButton: DDIPCTAButton
+    private let viwer: DDIPListViewer
```
🔴 오타가 있는 것 같아요!
```suggestion
    private let viewer: DDIPListViewer
```

kimkyunghun3 replied:
수정했습니다 감사합니다 👍

AhnSangHee commented on Projects/DesignSystem/Sources/Component/ListViwer/DDIPListViewer.swift:
```diff
@@ -12,10 +12,6 @@ public class DDIPListViewer: UIView {
-        didSet { self.viewerLabel.text = viewer }
-    }
-    
     private let viwerStackView: UIStackView = {
```
🔴 오타가 있어요!!
```suggestion
    private let viewerStackView: UIStackView = {
```

kimkyunghun3 replied:
앗 수정했습니다 확인감사합니다 👍

AhnSangHee commented on Projects/DesignSystem/Sources/Component/ListViwer/DDIPListViewer.swift:
```diff
@@ -12,10 +12,6 @@ public class DDIPListViewer: UIView {
     private var viewerLabel: UILabel = UILabel()
```
🟡 개인적으로 `DDIPListViewer`라는 네이밍이 무엇인지 알기 힘든 것 같아요! 응모자수를 나타내는 뷰라면 `DDipApplyViewer`와 같은 네이밍은 어떨까 조심스레 제안해봅니당!

kimkyunghun3 replied:
말씀들어보니 조금 더 applyViewer가 더 명확한거같네요! 다 수정했습니다~~

AhnSangHee reviewed (changes requested):
뷰가 복잡해서 구현하기 까다로웠을텐데 고생 많으셨습니다! 👍👍

ios-h commented on Projects/DesignSystem/Sources/Component/DeadlineCard/DDIPDeadlineView.swift:
```diff
@@ -0,0 +1,228 @@
+    public let imageIcon = UIImageView()
+    public let nameLabel = UILabel()
+    public let brandLabel = UILabel()
+    public let expirationLabel = UILabel()
```
❓ `유효기간` 이라는 라벨은 고정되어있고, 실제로 받는 유효기간 날짜 데이터는 모두 다르기 때문에 `유효기간` 자체를 표시하는 라벨과 유효기간 날짜를 받는 라벨의 2가지 라벨이 있어야 하지 않나 하는 생각이 듭니다..!!

AhnSangHee commented on Projects/DesignSystem/Sources/Component/DeadlineCard/DDIPDeadlineView.swift:
```diff
@@ -0,0 +1,228 @@
+    public let imageIcon = UIImageView()
+    public let nameLabel = UILabel()
+    public let brandLabel = UILabel()
+    public let expirationLabel = UILabel()
```
❓ `유효기간` 이라는 레이블은 고정되어있고, 실제로 받는 유효기간 날짜 데이터는 모두 다르기 때문에 `유효기간` 자체를 표시하는 레이블과 유효기간 날짜를 받는 레이블의 2가지 레이블이 있어야 하지 않나 하는 생각이 듭니다..! 근데 또 스타일 차이여서 그냥 레이블 하나로 해도 될 것 같기도 하네욤..!

kimkyunghun3 replied:
유효기간이라는 단어가 고정으로 되어있다면 추가로 라벨을 만들지 않아도 고정값으로 넘어주면 괜찮을 것 같습니다! 만약 이 값이 변동성이 있다면 라벨을 추가로 해야겠지만 현재 기획에서는 변동성이 없어보여서 일단 고정으로 넣어두겠습니다!!

AhnSangHee commented on Projects/DesignSystem/Sources/Component/ListCard/DDIPListCardView.swift:
```diff
@@ -0,0 +1,169 @@
+    
+    public let nameLabel = UILabel()
+    public let brandLabel = UILabel()
+    public let expirationLabel = UILabel()
```
❓요기도 앞에 코멘트 달았던 것과 동일하게, 2가지 레이블이 있어야 하지 않나 하는 생각이 듭니다! 요것도 스타일 차이기 때문에 넘기셔도 좋아요~!

kimkyunghun3 replied:
유효기간이라는 단어가 고정으로 되어있다면 추가로 라벨을 만들지 않아도 고정값으로 넘어주면 괜찮을 것 같습니다! 만약 이 값이 변동성이 있다면 라벨을 추가로 해야겠지만 현재 기획에서는 변동성이 없어보여서 일단 고정으로 넣어두겠습니다!!

AhnSangHee commented on Projects/DesignSystem/Sources/Component/ListCard/DDIPListCardView.swift:
```diff
@@ -0,0 +1,169 @@
+        return stackView
+    }()
+    
+    public let spaceLeftView: UIView = {
```
🟡  `spaceLeftView`, `spaceRightView` 라는 네이밍을 처음 보면 무엇인지 이해하기 어려울 것 같아요!! `leftSemicircleView`, `rightSemicircleView` (또는 `semicircleLeftView`, `semicircleRightView`..) 와 같이 더 명확한 네이밍은 어떨까 하는 제안을 조심스레 드려봅니다 ! 😮

kimkyunghun3 replied:
조금 더 명확하게 하려고 semiCircleSpaceLeftView, semiCircleSpaceRightView로 변경했습니다! 조언 감사합니다 :)

kimkyunghun3 committed:
:recycle: Viewer -> applyViewer로 네이밍 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle:  StackView+ -> UIStackView+ 로 폴더명 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: dashPattern 과 spacing, 3.5 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: spaceView -> semiCircleSpaceView

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 유효기간 단어 기본값으로 제공하도록 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

szzang-kr commented on Projects/DesignSystem/Sources/Component/Extension/UIStackView+.swift:
```diff
@@ -0,0 +1,17 @@
+
+import UIKit
+
+extension UIStackView {
+    func addArrangedSubviews(with views: [UIView]) {
+        views.forEach {
+            addArrangedSubview($0)
+        }
+    }
+}
```
UIView의 addSubView의 경우는 protocol을 채택해서 은닉화를 했는데,

UIStackView의 경우는 은닉화를 사용하지 않았네요. 어떤 이유때문일까요?

kimkyunghun3 replied:
먼저 StackView를 protocol로 은닉화 하려고 했는데, 사용처의 Self가 UIView여서 이 프로토콜을 채택하더라도 메서드를 사용할 수 없는 이슈가 있습니다.

그리고 두번째로는 StackView라고 범위를 한정지었기 때문에 StackView에서만 사용할 수 있으며 StackView의 성격상 여러 개의 뷰들을 더하기 때문에 이 메서드를 모두 사용할 것이라고 생각했습니다. 그리하여 protocol를 사용하지 않더라도 사용처의 범위가 좁아져 있기에 괜찮다고 판단했는데, 그럼에도 불구하고 채택하는 방향을 해야할까요?

szzang-kr replied:
굳이 채택하지 않아도 괜찮을것 같습니다.
어떠한 이유가 있는지 궁금했습니다. 😄

추가적으로 말씀해주신 첫번째 이슈를 해결하기 위해서는 UIStackView를 상속하고, protocol을 채택하는 새로운 BaseStackView를 만들어 사용해야합니다.

```swift
protocol AddArrangedSubviewsable where Self: UIStackView { }
class BaseStackView: UIStackView, AddArrangedSubviewsable { }
```

kimkyunghun3 replied:
아하 이렇게 해결할 수 있겠군요! 좋은 의견 감사합니다!!

szzang-kr commented on Projects/DesignSystem/Sources/Component/Extension/UIView+sugar.swift:
```diff
@@ -0,0 +1,23 @@
+        caShapeLayer.lineWidth = width
+        caShapeLayer.lineDashPattern = [3.5, 3.5]
+        let cgPath = CGMutablePath()
+        let cgPoint = [CGPoint(x: self.bounds.minX, y: 0), CGPoint(x: 310, y: 0)]
```
🟡 point를 고정으로 그리게 해뒀는데, 공통적으로 사용하는 method인 만큼 주입받아서 그리는 방식이 더 좋을것 같아요.

```swift
func createDottedLine(endPoint: CGPoint = .zero, end width: CGFloat, color: CGColor)
```

kimkyunghun3 replied:
이 부분에 대해서 주입받아서 해보았지만 결국 상위뷰에 대해 정확한 width를 확인하지 못해서
draw 메서드를 활용해서 DottedLine과 SpaceView를 구현 변경했습니다.
이렇게 하니 width를 억지로 만드는 것이 아니라, 동적으로 처리해줄 수 있어서 더 적절한 구현방식인듯싶습니다!

한번 확인해주시고 수정이 필요하다면 말씀해주세요 👍

szzang-kr reviewed (approved):
수고하셨습니다. 😄

kimkyunghun3 committed:
:recycle: ApplyViewer 사이즈 조정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: DashedLine draw method 활용하도록 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: SpaceView draw method 사용하도록 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ListCardView space, dashedLine 적용

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: SpaceView, Spacing 로직 변경에 따른 수정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: StackView extension 추가

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: StackView extension 통합에 따른 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성
